javascript/javascript1/week4/class-exercises/sentiment-analyzer.js

```diff
@@ -0,0 +1,87 @@
+const positiveWordsList = [
+  'cool',
+  'awesome',
+  'super',
+  'amazing',
+  'fantastic',
+  'brilliant',
+  'remarkable',
+  'incredible',
+  'spectacular',
+  'outstanding',
+  'exceptional',
+  'wonderful',
+  'magnificent',
+  'phenomenal',
+  'happy',
+];
+
+const negativeWordsList = [
+  'boring',
+  'sad',
+  'upset',
+  'terrible',
+  'awful',
+  'horrible',
+  'dismal',
+  'dreary',
+  'gloomy',
+  'disappointing',
+  'lackluster',
+  'mediocre',
+  'troublesome',
+  'annoying',
+  'distressing',
+];
+
+function getCommonWordsArray(sentenceArray, wordsList) {
+  const commonWords = [];
+  for (let i = 0; i < sentenceArray.length; i++) {
+    const word = sentenceArray[i];
+    if (wordsList.includes(word)) {
+      commonWords.push(word);
+    }
+  }
+  return commonWords;
+}
+
+function getSentimentScore(string) {
+  const wordsFromString = string.split(' ');
```

Comment:
Unfortunately this won't work, when you list words with comma. An alternative solution would be to use the `match` method and have 2 loops. One will be through the array of positive words, and another through the negative. Inside you will check if there is a match for the word.
It would look something like this:
```
if (sentence.match(listPositiveWords[i])) {
      positiveScoreWords.push(listPositiveWords[i]);
      score++;
    }
```

Reply:
Yasen, I agree with you that it's not working properly. However, I'm not entirely sure if I can use the match() method because the MDN documentation says that it retrieves the result of matching a string against a regular expression, and I don't have a regex. Could I just use includes() and go through two arrays of positive and negative words like you recommended?

```
function getCommonWordsArray(sentence, wordsList) {
  const commonWords = [];
  for (let i = 0; i < wordsList.length; i++) {
    const word = wordsList[i];
    if (sentence.includes(word)) {
      commonWords.push(word);
    }
  }
  return commonWords;
}
```